packages/enhanced/src/lib/container/runtime/FederationRuntimePlugin.ts

```diff
@@ -200,27 +208,72 @@ class FederationRuntimePlugin {
     }
   }
 
+  getDependency() {
+    if (this.federationRuntimeDependency)
+      return this.federationRuntimeDependency;
+    this.federationRuntimeDependency = new FederationRuntimeDependency(
+      this.getFilePath(),
+    );
+    return this.federationRuntimeDependency;
+  }
+
   prependEntry(compiler: Compiler) {
     if (!this.options?.virtualRuntimeEntry) {
       this.ensureFile();
     }
-    const entryFilePath = this.getFilePath();
-
-    modifyEntry({
-      compiler,
-      prependEntry: (entry) => {
-        Object.keys(entry).forEach((entryName) => {
-          const entryItem = entry[entryName];
-          if (!entryItem.import) {
-            // TODO: maybe set this variable as constant is better https://github.com/webpack/webpack/blob/main/lib/config/defaults.js#L176
-            entryItem.import = ['./src'];
-          }
-          if (!entryItem.import.includes(entryFilePath)) {
-            entryItem.import.unshift(entryFilePath);
-          }
-        });
-      },
-    });
+
+    //if using runtime experiment, use the new include method else patch entry
+    if (this.options?.experiments?.federationRuntime) {
+      compiler.hooks.thisCompilation.tap(
+        this.constructor.name,
+        (compilation: Compilation, { normalModuleFactory }) => {
+          const federationRuntimeDependency = this.getDependency();
+          const logger = compilation.getLogger('FederationRuntimePlugin');
+          const hooks =
+            FederationModulesPlugin.getCompilationHooks(compilation);
+          compilation.dependencyFactories.set(
+            FederationRuntimeDependency,
+            normalModuleFactory,
+          );
+          compilation.dependencyTemplates.set(
+            FederationRuntimeDependency,
+            new ModuleDependency.Template(),
+          );
+
+          compilation.addInclude(
```

Comment:
use add include to include the dependency across the build, this ensures its in all entry points